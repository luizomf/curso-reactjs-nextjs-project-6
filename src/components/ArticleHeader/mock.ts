import { ArticleHeaderProps } from '.';
import { data } from '../../api/dados.json';

const { title, excerpt, cover, author, categories, createdAt } = data.posts[0];

export default {
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt,
} as ArticleHeaderProps;
