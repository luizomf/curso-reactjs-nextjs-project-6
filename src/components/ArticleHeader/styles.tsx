import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
