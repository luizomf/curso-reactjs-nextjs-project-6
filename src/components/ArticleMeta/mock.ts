import { ArticleMetaProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'otavio-miranda',
    displayName: 'Ot√°vio Miranda',
  },
  categories: [
    {
      id: 'asdslgyh65',
      displayName: 'Tech',
      slug: 'tech',
    },
    {
      id: 'asdslgyh66',
      displayName: 'JS',
      slug: 'javascript',
    },
  ],
} as ArticleMetaProps;
