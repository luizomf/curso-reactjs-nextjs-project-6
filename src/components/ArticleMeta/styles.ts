import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }

    .categories span:last-child::after {
      content: '';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
