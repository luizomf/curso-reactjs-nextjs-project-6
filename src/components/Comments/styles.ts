import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacings.large} auto;
    padding: 0 ${theme.spacings.large};
  `}
`;
