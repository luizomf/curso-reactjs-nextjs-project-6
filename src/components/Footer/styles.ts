import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }
  `}
`;
