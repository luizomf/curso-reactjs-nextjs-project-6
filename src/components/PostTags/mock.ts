import { data } from '../../api/dados.json';

export default { tags: data.posts[0].tags };
