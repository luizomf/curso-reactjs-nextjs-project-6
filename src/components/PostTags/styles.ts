import styled, { css } from 'styled-components';

export const Wrapper = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} auto;
    max-width: ${theme.sizes.content};
    padding: 0 ${theme.spacings.large} 0;

    span {
      margin: 0 0 0 0.5rem;
    }

    span::before {
      content: ' ';
    }

    span::after {
      content: ', ';
    }

    span:last-child::after {
      content: '';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
