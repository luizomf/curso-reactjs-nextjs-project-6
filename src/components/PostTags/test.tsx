import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostTags, PostTagsProps } from '.';

import mock from './mock';

const props: PostTagsProps = mock;

describe('<PostTags />', () => {
  it('should render two tags', () => {
    renderTheme(<PostTags {...props} />);

    expect(screen.getByText(/Tags:/i)).toBeInTheDocument();
    expect(screen.getAllByRole('link')).toHaveLength(2);
  });

  it('should should match snapshot', () => {
    const { container } = renderTheme(<PostTags {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should should match snapshot with no tags', () => {
    const { container } = renderTheme(<PostTags {...props} tags={undefined} />);

    expect(container).toMatchSnapshot();
  });
});
