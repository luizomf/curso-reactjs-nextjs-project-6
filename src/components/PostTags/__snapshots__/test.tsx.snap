// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTags /> should should match snapshot 1`] = `
.c0 {
  margin: 2.4rem 0;
}

.c0 span {
  margin: 0 0 0 0.5rem;
}

.c0 span::before {
  content: ' ';
}

.c0 span::after {
  content: ', ';
}

.c0 span:last-child::after {
  content: '';
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

<div>
  <p
    class="c0"
  >
    tags:
    <span>
      <a
        href="/tag/use-effect"
      >
        useEffect
      </a>
    </span>
    <span>
      <a
        href="/tag/use-state"
      >
        useState
      </a>
    </span>
  </p>
</div>
`;
